cyral/resource.go

```diff
@@ -41,76 +48,112 @@ type ResourceOperationConfig struct {
 	NewResponseData func(d *schema.ResourceData) ResponseData
 }
 
+func CRUDResources(config []ResourceConfig) func(context.Context, *schema.ResourceData, interface{}) diag.Diagnostics {
+	return HandleRequest(config)
+}
+
 func CreateResource(createConfig, readConfig ResourceOperationConfig) schema.CreateContextFunc {
-	return HandleRequest(create, createConfig, &readConfig)
+	return HandleRequest(
+		[]ResourceConfig{
+			{
+				Type:            create,
+				OperationConfig: createConfig,
+			},
+			{
+				Type:            read,
+				OperationConfig: readConfig,
+			},
+		},
+	)
 }
 
 func ReadResource(readConfig ResourceOperationConfig) schema.ReadContextFunc {
-	return HandleRequest(read, readConfig, nil)
+	return HandleRequest(
+		[]ResourceConfig{
+			{
+				Type:            read,
+				OperationConfig: readConfig,
+			},
+		},
+	)
 }
 
 func UpdateResource(updateConfig, readConfig ResourceOperationConfig) schema.UpdateContextFunc {
-	return HandleRequest(update, updateConfig, &readConfig)
+	return HandleRequest(
+		[]ResourceConfig{
+			{
+				Type:            update,
+				OperationConfig: updateConfig,
+			},
+			{
+				Type:            read,
+				OperationConfig: readConfig,
+			},
+		},
+	)
 }
 
 func DeleteResource(deleteConfig ResourceOperationConfig) schema.DeleteContextFunc {
-	return HandleRequest(delete, deleteConfig, nil)
+	return HandleRequest(
+		[]ResourceConfig{
+			{
+				Type:            delete,
+				OperationConfig: deleteConfig,
+			},
+		},
+	)
 }
 
 func HandleRequest(
-	operationType string,
-	config ResourceOperationConfig,
-	consecutiveConfig *ResourceOperationConfig,
+	configSlice []ResourceConfig,
 ) func(context.Context, *schema.ResourceData, interface{}) diag.Diagnostics {
 	return func(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
-		log.Printf("[DEBUG] Init %s", config.Name)
-		c := m.(*client.Client)
-
-		var resourceData ResourceData
-		if config.NewResourceData != nil {
-			if resourceData = config.NewResourceData(); resourceData != nil {
-				if err := resourceData.ReadFromSchema(d); err != nil {
-					return createError(
-						fmt.Sprintf("Unable to %s resource", operationType),
-						err.Error(),
-					)
+		for _, config := range configSlice {
```

Comment:
What about assigning the `ResourceOperationConfig` to a `config` variable? This would keep the chained method calls simpler and also reduce the number of changes in this function.
```suggestion
		for _, resourceOperation := range resourceOperations {
		      config := resourceOperation.Config
```